ouroboros-consensus-cardano/src/shelley/Ouroboros/Consensus/Shelley/ShelleyHFC.hs

```diff
@@ -130,46 +130,20 @@ type ProtocolShelley = HardForkProtocol '[ ShelleyBlock (TPraos StandardCrypto)
 
 shelleyTransition ::
      forall era proto. ShelleyCompatible proto era
-  => PartialLedgerConfig (ShelleyBlock proto era)
-  -> Word16   -- ^ Next era's initial major protocol version
+  => Word16   -- ^ Next era's initial major protocol version
   -> LedgerState (ShelleyBlock proto era)
   -> Maybe EpochNo
-shelleyTransition ShelleyPartialLedgerConfig{..}
-                  transitionMajorVersionRaw
-                  state =
-      takeAny
-    . mapMaybe isTransition
-    . Shelley.Inspect.protocolUpdates genesis
-    $ state
+shelleyTransition transitionMajorVersionRaw =
+    isTransition . Shelley.Inspect.pparamsUpdate
   where
-    ShelleyTransitionInfo{..} = shelleyLedgerTransition state
 
-    -- 'shelleyLedgerConfig' contains a dummy 'EpochInfo' but this does not
-    -- matter for extracting the genesis config
-    genesis :: SL.ShelleyGenesis (EraCrypto era)
-    genesis = shelleyLedgerGenesis shelleyLedgerConfig
-
-    k :: Word64
-    k = SL.sgSecurityParam genesis
-
-    isTransition :: Shelley.Inspect.ProtocolUpdate era -> Maybe EpochNo
-    isTransition Shelley.Inspect.ProtocolUpdate{..} = do
-         SL.ProtVer major _minor <- proposalVersion
+    isTransition :: ShelleyLedgerUpdate era -> Maybe EpochNo
+    isTransition (ShelleyUpdatedPParams maybePParams newPParamsEpochNo) = do
+         SL.SJust pp <- Just maybePParams
+         let protVer = pp ^. SL.ppProtocolVersionL
          transitionMajorVersion <- SL.mkVersion transitionMajorVersionRaw
-         guard $ major == transitionMajorVersion
-         guard $ proposalReachedQuorum
-         guard $ shelleyAfterVoting >= fromIntegral k
-         return proposalEpoch
-       where
-         Shelley.Inspect.UpdateProposal{..} = protocolUpdateProposal
-         Shelley.Inspect.UpdateState{..}    = protocolUpdateState
-
-    -- In principle there could be multiple proposals that all change the
-    -- major protocol version. In practice this can't happen because each
-    -- delegate can only vote for one proposal, but the types don't guarantee
-    -- this. We don't need to worry about this, and just pick any of them.
-    takeAny :: [a] -> Maybe a
-    takeAny = listToMaybe
+         guard $ SL.pvMajor protVer == transitionMajorVersion
+         return newPParamsEpochNo
```

Comment:
`guard $ shelleyAfterVoting >= fromIntegral k` now seems to be missing

Reply:
I guess I misunderstood you. I thought you said it is no longer unnecessary.

Reply:
> I thought you said it is no longer unnecessary.

You probably mean "necessary"?

---

> I guess I misunderstood you.

Yeah, I could have made that more clear; when I commented in [this thread](https://github.com/IntersectMBO/ouroboros-consensus/pull/1078#discussion_r1587876479) on this code
```haskell
         -- We need to run this check first in order to ensure that PParams value is not
         -- forced unnecessarily. See documentation of `futurePParamsGovStateG` in ledger.
         guard $ shelleyAfterVoting >= fromIntegral k
```
I was just commenting on the fact that the comment does no longer apply, ie it is not relevant to run this check first to avoid forcing the PParams. The check is still necessary, but it is irrelevant if it runs before/after the `SL.pvMajor protVer == transitionMajorVersion` check. Does that make sense?

Reply:
> You probably mean "necessary"?

yes, that is what I meant :grin: 

From the name `shelleyAfterVoting` it sounded to me like a mark that indicates the two stability windows point before the end of the epoch, cause that is the only voting I am aware of in Shelley. That's why I inferred you comment as this guard now being redundant. Which led me to believe that it could now be redundant.
However, after reading the haddock a couple of times I realize that it means a number of blocks that was created after that point of two stability windows before the end of the epoch. In order to ensure that last block that potentially contained some votes can't be rolled back.

Which totally makes sense to me why we don't want to remove that guard. Brought it back.

Reply:
Perfect, thanks :+1: